autogpt_platform/frontend/src/tests/build.spec.ts

```diff
@@ -38,82 +84,108 @@ test.describe("Build", () => { //(1)!
-      if (block.name[0].toLowerCase() >= "m") {
-        continue;
-      }
-      if (!blockIdsToSkip.includes(block.id) && !blockTypesToSkip.includes(block.type)) {
-        await test.expect(buildPage.hasBlock(block)).resolves.toBeTruthy();
-      }
-    }
+  test("user can add blocks starting with d", async () => {
+    await addBlocksStartingWith("d");
+  });
 
-    // check that we can save the agent with all the blocks
-    await buildPage.saveAgent("all blocks test", "all blocks test");
-    // page should have a url like http://localhost:3000/build?flowID=f4f3a1da-cfb3-430f-a074-a455b047e340
-    await test.expect(page).toHaveURL(({ searchParams }) => !!searchParams.get("flowID"));
+  test("user can add blocks starting with e", async () => {
+    await addBlocksStartingWith("e");
   });
 
-  test.skip("user can add all blocks m-z", async ({ page }, testInfo) => {
-    // this test is slow af so we 100x the timeout (sorry future me)
-    test.setTimeout(testInfo.timeout * 100);
+  test("user can add blocks starting with f", async () => {
+    await addBlocksStartingWith("f");
+  });
 
-    await buildPage.closeTutorial();
-    await buildPage.openBlocksPanel();
-    const blocks = await buildPage.getBlocks();
+  test("user can add blocks starting with g", async () => {
+    await addBlocksStartingWith("g");
+  });
 
-    const blockIdsToSkip = await buildPage.getBlocksToSkip();
-    const blockTypesToSkip = ["Input", "Output", "Agent", "AI"];
-    console.log("⚠️ Skipping blocks:", blockIdsToSkip);
-    console.log("⚠️ Skipping block types:", blockTypesToSkip);
+  test("user can add blocks starting with h", async () => {
+    await addBlocksStartingWith("h");
+  });
 
-    // add all the blocks in order except for the agent executor block
-    for (const block of blocks) {
-      if (block.name[0].toLowerCase() < "m") {
-        continue;
-      }
-      if (!blockIdsToSkip.includes(block.id) && !blockTypesToSkip.includes(block.type)) {
-        await buildPage.addBlock(block);
-      }
-    }
-    await buildPage.closeBlocksPanel();
-    // check that all the blocks are visible
-    for (const block of blocks) {
-      if (block.name[0].toLowerCase() < "m") {
-        continue;
-      }
-      if (!blockIdsToSkip.includes(block.id) && !blockTypesToSkip.includes(block.type)) {
-        await test.expect(buildPage.hasBlock(block)).resolves.toBeTruthy();
-      }
-    }
+  test("user can add blocks starting with i", async () => {
+    await addBlocksStartingWith("i");
+  });
 
-    // check that we can save the agent with all the blocks
-    await buildPage.saveAgent("all blocks test", "all blocks test");
-    // page should have a url like http://localhost:3000/build?flowID=f4f3a1da-cfb3-430f-a074-a455b047e340
-    await test.expect(page).toHaveURL(({ searchParams }) => !!searchParams.get("flowID"));
+  test("user can add blocks starting with j", async () => {
+    await addBlocksStartingWith("j");
+  });
```

Comment:
We open a new browser, login, and test all the blocks with name starting with `j`, the next test does the same but starting with `k` and so on... 

We know the builder becomes slow UI-wise when having 30 or more blocks (which in turn makes the tests slower and requires waiting forever). 

This ensures, for now, we assert all blocks can render, but we never render more than +30 blocks at a time... It works for now, on the future we can find another solution if we soon end up with like 60 blocks starting with `A` 😮‍💨 😆

Once we refactor the builder and it can ran smooth with many blocks, we can come back here and undo these tests and run under a single one 💅🏽